Source/ProgressionSystemRuntime/Private/Components/PSSpotComponent.cpp

```diff
@@ -47,6 +47,50 @@ void UPSSpotComponent::OnReset_Implementation()
 	}
 }
 
+// Listen game states to switch character skin. 
+void UPSSpotComponent::OnGameStateChanged_Implementation(ECurrentGameState CurrentGameState)
+{
+	// it's possible that spot might not be loaded till that time so no ensure added
+	UPSSpotComponent* CurrentSpot = UPSWorldSubsystem::Get().GetCurrentSpot();
+
+	// check the spot  
+	if (CurrentSpot == nullptr || CurrentSpot != this)
+	{
+		return;
+	}
+
+	if (CurrentGameState == ECurrentGameState::GameStarting)
+	{
+		UMySkeletalMeshComponent& MeshComp = GetMeshChecked();
+		const int32 CurrentSkinIndex = MeshComp.GetAppliedSkinIndex();
+		bool isCurrentSkinAvailable = MeshComp.IsSkinAvailable(CurrentSkinIndex);
+
+		if (!isCurrentSkinAvailable)
+		{
+			for (int32 i = CurrentSkinIndex; i >= 0; i--)
+			{
+				if (i == 0)
+				{
+					isCurrentSkinAvailable = true;
+				}
+				else
+				{
+					isCurrentSkinAvailable = MeshComp.IsSkinAvailable(i);
+				}
+
+				if (isCurrentSkinAvailable)
```

Comment:
you can use `continue` here to decrease nesting, like:
```
if (!bIsCurrentSkinAvailable)
{
continue:
}
```

Reply:
done https://github.com/moinm3uw/ProgressionSystem/pull/63/commits/bddfe0eebe7d37db91e732207a5930b3034e9209